cpp/models/lct_secir/model.h

```diff
@@ -186,119 +176,95 @@ class Model
+            (1 - params.template get<DeathsPerCritical>()) * dummy;
+        dydt[LctState::template get_first_index<InfectionState::Dead>()] =
+            params.template get<DeathsPerCritical>() * dummy;
     }
 
     /**
-     * @brief Cumulates a simulation result with subcompartments to produce a result that divides the population only into the infection states defined in InfectionState.
+     * @brief Cumulates a simulation result with subcompartments to produce a result that divides the population only
+     *   into the infection states defined in InfectionState.
      *
-     * If the model is used for simulation, we will get a result in form of a TimeSeries with infection states divided in subcompartments.
+     * If the model is used for simulation, we will get a result in form of a TimeSeries with infection states divided 
+     * in subcompartments.
      * The function calculates a TimeSeries without subcompartmens from another TimeSeries with subcompartments. 
      * This is done by summing up the numbers in the subcompartments.
      * @param[in] result result of a simulation with the model.
      * @return result of the simulation divided in the Base infection states. 
      *  Returns TimeSeries with values -1 if calculation is not possible.
      */
-    TimeSeries<ScalarType> calculate_populations(const TimeSeries<ScalarType>& result) const
+    TimeSeries<ScalarType> calculate_compartments(const TimeSeries<ScalarType>& result) const
     {
         if (!(LctState::Count == result.get_num_elements())) {
             log_error("Result does not match infectionState of the Model.");
-            TimeSeries<ScalarType> populations((int)InfectionState::Count);
-            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant((int)InfectionState::Count, -1);
-            populations.add_time_point(-1, wrong_size);
-            return populations;
+            TimeSeries<ScalarType> compartments((Eigen::Index)InfectionState::Count);
+            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant(Eigen::Index(InfectionState::Count), -1);
+            compartments.add_time_point(-1, wrong_size);
+            return compartments;
         }
-        TimeSeries<ScalarType> populations((int)InfectionState::Count);
-        Eigen::VectorXd dummy((int)InfectionState::Count);
+        TimeSeries<ScalarType> compartments((Eigen::Index)InfectionState::Count);
+        Eigen::VectorXd dummy((Eigen::Index)InfectionState::Count);
         for (Eigen::Index i = 0; i < result.get_num_time_points(); ++i) {
             // Use segment of vector of the result with subcompartments of InfectionState with index j and sum up values of subcompartments.
-            dummy[(int)InfectionState::Susceptible] = result[i][0];
-            dummy[(int)InfectionState::Exposed] =
+            dummy[Eigen::Index(InfectionState::Susceptible)] = result[i][0];
```

Comment:
Here (and in the following accesses to ~Steve~ dummy) you use the ctor again, can you use casts instead?
Though it should be fine to use neither, the access operator should cause an implicit cast `size_t -> Index` (but it never hurts to make casts explicitly either).

Reply:
Yes you are absolutely right!